src/metrics/peer.go

```diff
@@ -109,6 +120,75 @@ func (pm *Peer) GetAllMessagesCount() uint64 {
 		pm.ProposerSlashing.Cnt)
 }
 
+// Fetch all the different metadata we received to save it into a new peer struct
+// TODO: Still few things to consider with new approach, like version handling
+// 		 or fetching the actual info with previous info from the peer
+func (pm *Peer) FetchHostInfo(hInfo BasicHostInfo) {
```

Comment:
not sure i understand this function. imho `peer.go` should be as dummy as possible and i have the feeling it adds some very specific things

Reply:
I wanted to discuss this implementation with you. I was perhaps going further than I should, but my idea was to prepare the function and the metadata to track possible versions in the future. This struct also summarizes all the information that we can stract from a remote peer by using libp2p identify requests.
This way, we could track this basic versioned info about each remote hosts. Perhaps all this libp2p host-related info could be stored in the Peer struct directly